src/parser/utils.ts

```diff
@@ -58,7 +58,8 @@ export function haveBuffer(): boolean {
 
 /** Callable in any environment to check if value is a Buffer */
 export function isBuffer(value: unknown): value is Buffer {
-  return typeof value === 'object' && value?.constructor?.name === 'Buffer';
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return isUint8Array(value) && typeof (value as any)?.toJSON === 'function';
 }
```

Comment:
I think it would be better to check for the properties on the buffer that the driver actually needs to exist in order to successfully do its job instead of relying on an arbitrary condition to infer that the object is of the correct type; furthermore, we should add some unit tests around types that are expected to pass the check and the ones that are expected to fail

Reply:
> I think it would be better to check for the properties on the buffer that the driver actually needs to exist in order to successfully do its job instead of relying on an arbitrary condition to infer that the object is of the correct type;

Soooo... if clean code is the goal here, the way to do that is to just never check whether something is a Buffer, because there's never a good reason to do that (e.g. all Node.js built-in APIs which accept a Buffer also accept all other kinds of Uint8Arrays). Luckily, that's actually pretty easy to do (bonus bugfix included 🙂): https://github.com/mongodb/js-bson/pull/432

Reply:
Yeah that was pretty much my thought, will take a look at the new code soon, thanks!